plugins/modules/aws_acm.py

```diff
@@ -234,11 +267,73 @@
 '''
 
 
+import base64
+from copy import deepcopy
+import re  # regex library
+
+try:
+    import botocore
+except ImportError:
+    pass  # handled by AnsibleAWSModule
+
 from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
 from ansible_collections.amazon.aws.plugins.module_utils.acm import ACMServiceManager
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import compare_aws_tags
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import (
+    boto3_tag_list_to_ansible_dict,
+    ansible_dict_to_boto3_tag_list,
+)
 from ansible.module_utils._text import to_text
-import base64
-import re  # regex library
+
+
+def ensure_tags(client, module, resource_arn, existing_tags, tags, purge_tags):
+    if tags is None:
+        return (False, existing_tags)
+
+    tags_to_add, tags_to_remove = compare_aws_tags(existing_tags, tags, purge_tags)
+    changed = bool(tags_to_add or tags_to_remove)
+    if tags_to_add:
+        if module.check_mode:
+            module.exit_json(
```

Comment:
Imo the module should not exit in check mode asap.  
It might return usefull information in even in check_mode: https://github.com/ansible-collections/community.aws/blob/main/plugins/modules/aws_acm.py#L375-L387

Reply:
The example you provide does exactly that. The only difference is instead having an if check_mode: exit, it has if not chec_mode: do X else exit.